### PR TITLE
fix: fixed arg validation for format

### PR DESCRIPTION
**Pull request by anisatahlil**
Issue #, if available:

**Description of Changes**
Argument checks for `agc configure format` was not handled gracefully. This caused stack error to output into the console.
```
$ agc configure format
panic: runtime error: index out of range [0] with length 0
```

[//]: #  (A description of the change that you made and the new user experience that it creates)

**Description of how you validated changes**
```
$ agc configure format
Error: format value must be provided
```
[//]: #  (A description of you validated your changes and any relevant logs that show your change works)

**Checklist**

- [x] If this change would make any existing documentation invalid, I have included those updates within this PR
- [x] I have added unit tests that prove my fix is effective or that my feature works
- [x] I have linted my code before raising the PR
- [x] Title of this Pull Request follows Conventional Commits standard: https://www.conventionalcommits.org/en/v1.0.0/

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/cli/internal/pkg/cli/configure_format.go [hunk 1]
```diff
@@ -56,14 +56,14 @@ func BuildConfigureFormatCommand() *cobra.Command {
 		Short: "Sets default format option for output display of AGC commands. Valid format options are 'text' and 'table'",
 		Args:  cobra.ArbitraryArgs,
```
Comment (AndreyDovydenko) [inline packages/cli/internal/pkg/cli/configure_format.go:57]: Should this be `Args:  cobra.ExactArgs(1),` to ensure the single argument passed. or is there a case where we would want to have more than one arg passed to this command?

File: packages/cli/internal/pkg/cli/configure_format.go [hunk 2]
```diff
@@ -30,8 +30,8 @@ func newFormatContextOpts(vars formatContextVars) (*formatContextOpts, error) {
 	}, nil
 }
 
-func (o *formatContextOpts) Validate() error {
-	if o.formatContextVars.format == "" {
+func (o *formatContextOpts) Validate(args []string) error {
```
Comment (AndreyDovydenko) [inline packages/cli/internal/pkg/cli/configure_format.go:33]: Should we be passing an array in here if we're only reading `args[0]` out of it later on?

**Review by AndreyDovydenko — CHANGES_REQUESTED** [review]

File: packages/cli/internal/pkg/cli/configure_format.go (hunk 1 above)
Comment (anisatahlil) [reply packages/cli/internal/pkg/cli/configure_format.go:57]: The feedback from pervious PRs suggested to move away from `Args: cobra.ExactArgs(1)` to give customers a more readable error when argument is not provided.

File: packages/cli/internal/pkg/cli/configure_format.go (hunk 2 above)
Comment (anisatahlil) [reply packages/cli/internal/pkg/cli/configure_format.go:33]: That function is handling validation which also checks if the array is empty to provide users with a readable error message

File: packages/cli/internal/pkg/cli/configure_format.go [hunk 3]
```diff
@@ -30,8 +30,8 @@ func newFormatContextOpts(vars formatContextVars) (*formatContextOpts, error) {
...
+	if len(args) == 0 {
```
Comment (elliot-smith) [inline packages/cli/internal/pkg/cli/configure_format.go:34]: This should be != 1 with a message similar to "a single format value must be provided"

File: packages/cli/internal/pkg/cli/configure_format.go (hunk 1 above)
Comment (elliot-smith) [reply packages/cli/internal/pkg/cli/configure_format.go:57]: "Error: accepts 1 arg(s), received 0" only tells the customer the what while "Error: a single format value must be provided" tells the customer the why as well. This means that for a customer to resolve the issue, they don't need to type "-h" to know what the values are.

Ideally, we could populated the valid args like you mention or generate the message with the valid options like we do with some of our other commands.

**Review by elliot-smith — APPROVED** [review]

**Pull request merged**